app/objective/get.controller.js

```diff
@@ -9,18 +9,64 @@ const getObjective = async (
   const nexturl = path.substring(0, path.lastIndexOf('/'))
   const backurl = removeUrlLevels(path, 2)
   const renderDetails = { ...renderInfo, nexturl, backurl }
+  let displayNeeds
+  let displayObjective = { description: '', needs: [] }
 
+  // get the saved or submitted objective
   try {
     if (body.objective !== undefined) {
-      renderDetails.objective = body.objective
+      displayObjective.description = body.objective
+      displayObjective.needs = body.needs || []
     } else if (objectiveId.toLowerCase() !== 'new') {
-      renderDetails.objective = await getSentencePlanObjective(planId, objectiveId, token)
+      const savedObjective = await getSentencePlanObjective(planId, objectiveId, token)
+
+      // convert need to an array if only one has been passed back
+      if (savedObjective.needs) {
+        if (!(savedObjective.needs instanceof Array)) {
+          savedObjective.needs = [savedObjective.needs]
+        }
+      }
+      displayObjective = savedObjective
     }
-    return res.render(`${__dirname}/index`, { ...body, errors, errorSummary, ...renderDetails })
   } catch (error) {
     logger.error(`Could not retrieve objective ${objectiveId} for sentence plan ${planId}, error: ${error}`)
     return res.render('app/error', { error })
   }
+
+  // get all the needs that apply to this sentence plan
+  try {
+    const needs = await getSentencePlanNeeds(planId, token)
+    // convert to format for display
+    displayNeeds = needs
+      .map(need => {
```

Comment:
Destructure here and then you can tidy up the object declaration below.
So...
({id:value, description: html, active, riskOfHarm=false}) =>